### PR TITLE
majima: init at 0.4.0

### PR DESCRIPTION
**Pull request by ufUNnxagpM**
## Description of changes

majima is a rust cli tool for generating random usernames in various formats

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -5315,6 +5315,8 @@ with pkgs;
 
   maigret = callPackage ../tools/security/maigret { };
 
+  majima = callPackage ../tools/security/majima { };
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:5318]: Use the pkgs/by-name hierarchy

File: pkgs/tools/security/majima/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{ lib, fetchFromGitLab, rustPlatform }:
+rustPlatform.buildRustPackage rec {
+  pname = "majima";
+  version = "0.4.0";
+
+  src = fetchFromGitLab {
+    owner = "gumball-overall";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/tools/security/majima/default.nix:8]:
```suggestion
    repo = "majima";
```

File: pkgs/tools/security/majima/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    rev = "4ffe01b9b4a92b401ee1c76c5cefe1ba0f0812de";
```
Comment (AndersonTorres) [inline pkgs/tools/security/majima/default.nix:9]: Use a revision string based on version.

File: pkgs/tools/security/majima/default.nix [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    sha256 = "sha256-S62DQfvZFg8C26YG+fIVJj5cJ6mz73JXSgdu5yoK0Yo=";
```
Comment (AndersonTorres) [inline pkgs/tools/security/majima/default.nix:10]:
```suggestion
    hash = "sha256-S62DQfvZFg8C26YG+fIVJj5cJ6mz73JXSgdu5yoK0Yo=";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (ufUNnxagpM) [reply pkgs/top-level/all-packages.nix]: Fixed in the latest push

File: pkgs/tools/security/majima/default.nix (hunk 3 above)
Comment (ufUNnxagpM) [reply pkgs/tools/security/majima/default.nix]: Fixed in the latest push

File: pkgs/by-name/ma/majima/package.nix [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    owner = "gumball-overall";
+    repo = "majima";
+    rev = version;
+    hash = "sha256-S62DQfvZFg8C26YG+fIVJj5cJ6mz73JXSgdu5yoK0Yo=";
+  };
+
+  cargoHash = "sha256-zMQO6McnnGbp52A9n/h6yZTU9VH7vak2TSP0HLqDlKw=";
+
+  meta = {
+    description = "Generate random usernames quickly and in various formats.";
```
Comment (AndersonTorres) [inline pkgs/by-name/ma/majima/package.nix:16]:
```suggestion
    description = "Generate random usernames quickly and in various formats";
```

File: pkgs/by-name/ma/majima/package.nix [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    homepage = "https://majima.matte.fyi/";
+    maintainers = [ lib.maintainers.ufUNnxagpM ];
+    license = lib.licenses.mit;
+  };
```
Comment (AndersonTorres) [inline pkgs/by-name/ma/majima/package.nix:20]: Set meta.mainProgram.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**